benchmarks/nel/scripts/datasets/dataset.py

```diff
@@ -220,61 +249,86 @@ def evaluate(
         candidate_generation: bool = True,
         baseline: bool = True,
         context: bool = True,
+        spacyfishing: bool = True,
         n_items: Optional[int] = None,
     ) -> None:
         """Evaluates trained pipeline on test set.
         baseline (bool): Whether to include baseline results in evaluation.
         context (bool): Whether to include the local context in the model.
+        spacyfishing (bool): Whether to include evaluation with spacyfishing.
         n_items (Optional[int]): How many items to consider in evaluation. If None, all items in test set are used.
         """
 
         # Load resources.
         self._load_resource("nlp_best")
+        self._load_resource("nlp_base")
         self._load_resource("kb")
+
+        # Compile test set.
         test_set_path = self._paths["corpora"] / "test.spacy"
         with open(test_set_path, "rb"):
-            test_set = [
-                Example(self._nlp_best(doc.text), doc)
-                for doc in DocBin()
-                .from_disk(test_set_path)
-                .get_docs(self._nlp_best.vocab)
-            ]
+            test_set: List[Example] = []
+            for doc in DocBin().from_disk(test_set_path).get_docs(self._nlp_best.vocab):
+                predicted_doc = self._nlp_best(doc.text)
+                ents: List[Span] = []
+                for ent in predicted_doc.ents:
+                    # spaCy includes leading articles in entities, our benchmark datasets don't. Hence we drop all
+                    # leading "the " and adjust the entity positions accordingly.
+                    ents.append(
+                        doc.char_span(ent.start_char + 4, ent.end_char, label=ent.label, kb_id=ent.kb_id)
+                        if ent.text.lower().startswith("the ") else ent
+                    )
+                predicted_doc.ents = ents
+                test_set.append(Example(predicted_doc, doc))
+
         self._nlp_best.config["incl_prior"] = False
+        if spacyfishing:
+            self._nlp_base.add_pipe("entityfishing", last=True)
 
         # Evaluation loop.
         label_counts = dict()
         cand_gen_label_counts = defaultdict(int)
         baseline_results = evaluation.DisambiguationBaselineResults()
         context_results = evaluation.EvaluationResults("Context only")
         combo_results = evaluation.EvaluationResults("Context and Prior")
+        spacyfishing_results = evaluation.EvaluationResults("spacyfishing")
         candidate_results = evaluation.EvaluationResults("Candidate gen.")
 
         for example in tqdm.tqdm(
             test_set, total=n_items, leave=False, desc="Processing test set"
         ):
+            example: Example
             if len(example) > 0:
-                correct_ents = {
+                entity_linker: EntityLinker_v1 = self._nlp_best.components[  # type: ignore
+                    self._nlp_best.component_names.index("entity_linker")
```

Comment:
Lets move this index to a separate var so that hopefully the declaration of `entity_linker` can be one line instead.

Reply:
Dunno why it took me so long to learn about `.get_pipe()` :smile: